### PR TITLE
Add support for aggregations

### PR DESCRIPTION
**Pull request by vedarthk**
### Description
Elasticsearch has deprecated facets and aggregrations are to be used hence forth. These changes will add support for aggregations and will keep facets as well. Facets will be deprecated in coming versions.

Aggregations Reference:
https://www.elastic.co/guide/en/elasticsearch/reference/1.5/search-aggregations.html

Facets Deprecation:
https://www.elastic.co/guide/en/elasticsearch/reference/1.5/search-facets.html

### Query Changes
There are no major changes in query, to make aggregations query `facets` needs to replaced by `aggregations` eg:

Old facets query
```
type:candidates (name:"John Doe" starred:true) (python or java) facets:[location]
```

New aggregations query
```
type:candidates (name:"John Doe" starred:true) (python or java) aggregations:[location]
```

Generated Query:
```json
{
    "sort": [
        {
            "created_on": "desc"
        }
    ],
    "query": {
        "filtered": {
            "filter": {
                "bool": {
                    "should": [],
                    "must_not": [],
                    "must": [
                        {
                            "type": {
                                "value": "candidates"
                            }
                        },
                        {
                            "term": {
                                "client_id": 1
                            }
                        },
                        {
                            "term": {
                                "user_id": 2
                            }
                        }
                    ]
                }
            },
            "query": {
                "query_string": {
                    "query": "(name:\"John Doe\" starred:true) (python OR java)",
                    "default_operator": "and"
                }
            }
        }
    },
    "aggregations": {
        "location": {
            "aggregations": {
                "location": {
                    "terms": {
                        "field": "location_nonngram",
                        "size": 20
                    }
                }
            }
        }
    }
}
```

### CODE REVIEW

**Comment by bIgBV** [comment]
@vedarthk why is the `aggreagtions` object in the generated query nested when it is a top level mapping?

File: plasticparser/grammar_parsers.py [hunk 1]
```diff
@@ -15,6 +15,14 @@ def get_query(self):
         return self.facets_dsl
 
 
+class Aggregations(object):
+    def __init__(self, facets_dsl):
+        self.facets_dsl = facets_dsl
```
Comment (uttamk) [inline plasticparser/grammar_parsers.py]: sorry for the nitpick... can we rename it to aggregations_dsl ?

File: plasticparser/grammar_parsers.py [hunk 2]
```diff
@@ -180,13 +199,80 @@ def parse_single_facet_expression(tokens):
...
+    """
+    Parses single aggregation query. Following is example input and output:
+
+    INPUT
+    type:candidates (name:"John Doe" starred:true) (python or java) facets:[location]
+
+    OUTPUT
+    {
+        ...
+        "aggregations": {
```
Comment (uttamk) [inline plasticparser/grammar_parsers.py]: Same question that @bIgBV asked ? Does the ES query need 2 nested aggregations ? Just curious

File: circle.yml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+test:
```
Comment (uttamk) [inline circle.yml]: Running on Circle ... whatever happened to Travis

**Comment by uttamk** [comment]
@vedarthk overall, this should not affect the existing facets ... Awesome job strangling the existing ES version ... Looks good to merge ... just make sure you test this version on local to see that existing queries are not broken
cc. @anantkpal

**Comment by vedarthk** [comment]
Removal of sub aggregations will be done in next minor upgrade.

**Comment by uttamk** [comment]
You mean ES minor upgrade @vedarthk or plastic parser ?

**Pull request merged**